pageserver/src/tenant/vectored_blob_io.rs

```diff
@@ -157,15 +163,23 @@ pub struct VectoredReadPlanner {
     // Arguments for previous blob passed into [`VectoredReadPlanner::handle`]
     prev: Option<(Key, Lsn, u64, BlobFlag)>,
 
-    max_read_size: usize,
+    max_read_size: Option<usize>,
 }
 
 impl VectoredReadPlanner {
     pub fn new(max_read_size: usize) -> Self {
         Self {
             blobs: BTreeMap::new(),
             prev: None,
-            max_read_size,
+            max_read_size: Some(max_read_size),
+        }
+    }
+
+    pub fn new_without_max_limit() -> Self {
```

Comment:
Why do we need this? I'm strongly against unbounded vectored reads for a variety of reasons, most of them related to predictability of peak memory usage.

Reply:
Ahh, I now read the comment in the caller. 

Maybe let's

1. dupe some of the comment in the caller as a doc comment here
1. rename this method to `new_caller_controlled_max_limit` to set the right tone

Reply:
resolved in https://github.com/neondatabase/neon/pull/8006/commits/1ef86d26d6f59092a73b0669cdec7d87fb1f29af